text/XXXX-delete_and_drop.md

```diff
@@ -0,0 +1,225 @@
+
+A better UX for removing data from qri should clearly answer two common "uh oh" scenarios a user is likely to encounter:
+
+1. "I made a mistake while saving"
+2. "I want to undo a push"
+
+The first is an example of removing data from history. The second is an example of removing data from the network. In both scenarios the user is a data author, and a good answer to these questions should present well-documented, easy-to-find commands that when executed leave the user feeling that they're in a clean state. 
+
+A well-documented desctrive command should be clearly labeled as the "opposite" of a constructive command and present warnings _before_ execution. Once executed, commands remove data should show as few caveat/errors/warnings as possible, or at least connect caveats to a clear next course of action. A feeling of "clean state" comes when vommands like `qri log`, `qri list` and `qri status` match user expectations after running a destructive command.
+
+To improve our "undo UX", this RFC proposes three changes to qri CLI:
+1. rename `remove` to `delete`, clarify its text to talk about history instead of just storage
+2. Add an `amend` flag on the `save` command that _replaces_ the HEAD commit
+3. Add a new top-level command: `drop`, for dropping datasets you don't own
+
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+### Renaming Remove
+Here's the current (qri v0.9.8) description of the `remove` command:
+
+```
+$ qri remove --help
+Remove gets rid of a dataset from your qri node. After running remove, qri will
+no longer list your dataset as being available locally. 
+[rambling bit about IPFS storage redacted]
+```
+
+This text is misleading. It seems to be about storage, when the primary purpose of  `remove` is to edit history. Storage savings are a byproduct of editing one or more commits out of a history. This RFC sets aside the tight coupling between history and storage for another day, and instead focueses on fixing the role of remove by clearifying it's documentation:
+
+```
+$ qri delete --help
+Delete removes commits from a dataset history. Use delete to "correct the record", 
+removing commits in a dataset history. Delete always works from the latest (HEAD) 
+commit, working backwards toward the first commit. To delete from a dataset you 
+must be able to edit it.
+
+Delete requires a "revisions" flag, specifying
+
+To delete an entire dataset and free up the name again, use `--all` (or a 
+number of revisions longer than the history itself). Before doing this be sure
+to drop your dataset from other remotes
+
+Usage:
+  qri delete [DATASET] [--revisions|--all] [flags]
+
+Examples:
+  # delete the latest commit from annual_pop
+  $ qri delete me/annual_pop --revisions 1
+
+  # delete the latest two versions from history, keeping the underlying data
+  $ qri delete me/annual_pop --revisions 2  --keep-storage
```

Comment:
If the user doesn't use the `--keep-storage` flag, then we purge blocks from IPFS? This seem hard to do, and dangerous, and prevents any kind of "undo" functionality like what I mentioned above. It also makes us work very differently than git. I would rather that we rely on the `storage` command to run "gc" when the user wants it. Or at least, have the dangerous operation be *opt-in*, not opt-out, meaning a `--purge` flag instead.